### ISSUE TITLE
Faster R-CNN: Towards Real-Time Object Detection with Region Proposal Networks

### ISSUE THREAD

**Issue by Yagami360** [Yagami360/MachineLearning-Papers_Survey#76]
## 0. 論文情報・リンク
- 論文リンク：https://arxiv.org/abs/1506.01497
- 公開日時：2015/06/04
- 組織 : Microsoft Research
- 被引用数（記事作成時点）：19681 件
- 実装コード（Caffle）：https://github.com/rbgirshick/py-faster-rcnn
- Publication : 

## 1. どんなもの？
- Fast R-CNN での RoI 領域検出部分も全畳み込みネットワークである Region Proposal Network (RPN) で行うようにすることで、処理速度と精度を高めた物体検出モデル。現在多くの物体検出モデルやセグメンテーションモデルのベースラインとして採用されている。

## 2. 先行研究と比べてどこがすごいの？
- 既存の物体検出モデルである Fast R-CNN では、１枚の入力画像に対して１度の畳み込み処理のみで機能するようにすることで、R-CNN と比較して大幅な高速化と精度向上を実現した。一方で、RoI 領域検出部分は依然として Selective Search（SS）などの非DNNの古典的手法を使っており、この部分が処理速度のボトルネックになっている。
本手法ではこの RoI 領域の検出部分も、CNN で行うようにした Region Proposal Network (RPN) を提案し、高速化と精度向上の両方を実現している。
- また、RoI 領域検出部分も CNN で行うことで、物体検出ネットワーク全体を end2end で学習出来るようにし、さらなる精度向上を実現している。
- PASCAL VOC での物体検出で、当時のSOTAを達成

## 3. 技術や手法の"キモ"はどこにある？
- アーキテクチャの全体像
    <image src="https://user-images.githubusercontent.com/25688193/86593576-77121d00-bfd0-11ea-8e8d-3876743a484f.png" width="443">
	上図は Faster R-CNN のアーキテクチャの全体像を示した図である。
	本ネットワークは、以下の２つのネットワークから構成されている。
	
	a. Region Proposal Network (RPN)
		入力画像の RoI 領域を検出するためのネットワーク。
		本手法が新たに提案しているのは、この部分のネットワークである。
		 
	b. Fast R-CNN の検出器
		RPN での RoI 領域検出より後段のネットワークは、Fast R-CNN における RoI 検出処理部分以外のアーキテクチャを世襲している。
		より詳細には、RoI 領域を集約する RoI pooling 層、及び softmax 出力（＝物体予想ラベルの確率値）とBBOXオフセット出力のためのFC層で構成される
		
	これらのネットワーク全体は、ImageNet で事前学習済み CNN モデル（VGG等）を fine-tuning したネットワークで構成され、end2end に学習される。

- Region Proposal Networks [RPN]
    <image src="https://user-images.githubusercontent.com/25688193/86593693-a759bb80-bfd0-11ea-818b-c974309deb08.png" width="760">

	Region Proposal Networks [RPN] では、畳み込み層を使って、RoI 領域の検出を行う。
	具体的には、以下のような処理を行う。
	※ アンカーの概念により、様々な形状での物体検出を可能にしているところに注目。

	1. 入力画像を conv 処理し、入力画像の特徴マップを生成する。
	2. 入力画像の特徴マップ上で、3x3 の siling windows（畳み込みフィルター）を走らせ、3x3x256 の中間特徴マップを生成する。
		この際に、固定長の siling windows では様々な形状の物体領域に対して対応できないので、右上図のように、siling windows の中心点で定義される Anchors（アンカー） と１つの Anchor から複数個得られる Anchor boxes の概念を導入することで、様々な形状の物体検出に対応できるようにする。
		この Anchor boxes は、１つの Anchor （siling windows の中心座標）に対して、スケール値（64, 128, 256 pixels の３つ）と縦横アスペクト比（1:1, 1:2, 2:1の３つ）の情報を持つ９つ（k=3x3=9）の長方形ボックスである。
		
	3. 3x3x256 の特徴マップを、以下の２つの出力に回帰する。
		① k 個のAnchors boxes に対しての、物体 or 背景の分類スコア（2k個）
		② k 個のAnchors boxes に対しての、Anchor boxes 短形値（4k個）

- 損失関数
	Faster R-CNN での損失関数は、Fash R-CNN と同様に、Multi-task loss で定義される。
	但し、アンカーを導入したため、以下の式のように各アンカーに対して計算される損失関数となる。
ネットワークはこの損失関数に従って end2end に学習される。
    <image src="https://user-images.githubusercontent.com/25688193/86593778-d40dd300-bfd0-11ea-9b7f-aee5ecb20104.png" width="720">

- RPN と Fast R-CNN 間でのネットワーク重みの共有
    <image src="" width="">

## 4. どうやって有効だと検証した？
- 既存の RoI 領域検出モデルとの物体検出精度の定量比較検証
    <image src="https://user-images.githubusercontent.com/25688193/86593848-f7d11900-bfd0-11ea-886b-a941ca422c67.png" width="570">
	既存の RoI 領域検出方法である SS [Selective Search] よりも、本手法が提案している RPN で RoI 検出することで、物体検出精度が向上していることが見て取れる

- 既存の RoI 領域検出モデルとの処理速度の定量比較検証
    <image src="https://user-images.githubusercontent.com/25688193/86594008-3cf54b00-bfd1-11ea-91f1-a85ab03f15eb.png" width="630">
    既存の RoI 領域検出方法である SS [Selective Search] よりも、本手法が提案している RPN で RoI 検出することで、処理速度が大きく向上していることが見て取れる（約１０倍）

## 5. 議論はあるか？
- xxx

## 6. 次に読むべき論文はあるか？
- xxx

## 7. 参考文献
- [Fast R-CNN](https://github.com/Yagami360/MachineLearning-Papers_Survey/issues/75)
